rust/src/utility/compression.rs

```diff
@@ -0,0 +1,17 @@
+use flate2::{read::GzDecoder, write::GzEncoder, Compression};
+use std::io::prelude::*;
+
+/// Compress bytes with gzip
+pub fn compress_gzip(bytes: &mut [u8]) -> anyhow::Result<()> {
+    let mut encoder = GzEncoder::new(Vec::new(), Compression::default());
+    Ok(encoder.write_all(bytes)?)
+}
+
+/// Decompress gzip compressed bytes
+pub(crate) fn decompress_gzip(bytes: &[u8]) -> anyhow::Result<Vec<u8>> {
+    let mut buf = vec![];
+    let mut gz = GzDecoder::new(bytes);
+
+    gz.read_to_end(&mut buf)?;
+    Ok(buf)
+}
```

Comment:
(not-blocking; FWIW): there is already code to `tar` a file.

https://github.com/Granola-Team/mina-indexer/blob/7f3efe4cb7b22f811cf5d2b005acb5f2f6331e31/rust/src/store/mod.rs#L333-L363

Reply:
Yes, but we need to compress the file, not tar it